### DISCUSSION TITLE
How to load Leaflet plugins?

### DISCUSSION THREAD

**Discussion by FeldrinH** [zauberzeug/nicegui#2795] [Q&A]
### Question

I am trying to use Leaflet plugins with nicegui (specifically https://github.com/coryasilva/Leaflet.ExtraMarkers and https://github.com/bbecquet/Leaflet.RotatedMarker, but as far as I know this problem applies to all Leaflet plugins that use JavaScript).

The problem I have is that I can't figure out how to load the JavaScript part of the plugins at the right time.
The plugins need to be loaded (as far as I can tell) after the Leaflet global (`L`) is initialized, but before any of the markers using the plugins are added to the map.

If I use `ui.add_head_html` or `ui.add_body_html` to add script tags for the plugins then the plugins are loaded too early and fail because `L` is not defined.

What is the correct approach here? Does nicegui support using Leaflet plugins?

**Comment by falkoschindler** [comment]
Hi @FeldrinH,
So far we haven't worked with Leaflet plugins. I'm not sure how easy they can be integrated. Ideas and proofs of concept are very welcome! 🙂
See https://github.com/zauberzeug/nicegui/discussions/2767 for a related feature request.
